src/__tests__/optimize.test.ts

```diff
@@ -16,60 +14,42 @@ for (let i = 0; i < nbPoints; i++) {
   x[i] = (i - nbPoints / 2) * xFactor;
 }
 
-describe('Optimize sum of Lorentzian', () => {
+// closing issue 80
+
+describe('Optimize sum of Lorentzians', () => {
   const peaks = [
+    { x: -0.5, y: 1, fwhm: 0.05, shape: { kind: 'lorentzian' } },
+    { x: 0.5, y: 1, fwhm: 0.05, shape: { kind: 'lorentzian' } },
+  ];
+  const peaksGenerator = [
     { x: -0.5, y: 1, fwhm: 0.05 },
     { x: 0.5, y: 1, fwhm: 0.05 },
   ];
-  const data: DataXY = generateSpectrum(peaks, {
+
+  const data: DataXY = generateSpectrum(peaksGenerator, {
     generator: {
       from: -1,
       to: 1,
       nbPoints: 101,
     },
-    peakOptions: {
-      factor: 6,
-    },
   });
 
-  it('group of two GL', () => {
-    let pTrue = [-0.5, 0.5, 0.001, 0.001, 0.31, 0.31];
-    let yData = sumOfLorentzians(pTrue);
-    let peakList = [
-      { x: -0.5, y: 0.0009, fwhm: (xFactor * nbPoints) / 8 },
-      { x: 0.52, y: 0.0009, fwhm: (xFactor * nbPoints) / 8 },
-    ];
-    let result = optimize({ x, y: x.map((i) => yData(i)) }, peakList, {
-      shape: { kind: 'lorentzian' },
-    });
-    let nL = pTrue.length / 3;
-    for (let i = 0; i < nL; i++) {
-      let pFit = result.peaks[i];
-      expect(pFit.x).toBeCloseTo(pTrue[i], 2);
-      expect(pFit.y).toBeCloseTo(pTrue[i + nL], 2);
-      expect(pFit.fwhm).toBeCloseTo(pTrue[i + nL * 2], 1);
-      expect(peakList).toStrictEqual([
-        { x: -0.5, y: 0.0009, fwhm: (xFactor * nbPoints) / 8 },
-        { x: 0.52, y: 0.0009, fwhm: (xFactor * nbPoints) / 8 },
-      ]);
-    }
-  });
-
-  // ADDED
   it('positive maxima peaks', () => {
     let peakList = [
       {
-        x: -0.5,
-        y: 1,
+        x: -0.52,
+        y: 0.9,
         fwhm: 0.08,
+        shape: { kind: 'lorentzian' } as Shape1D,
       },
       {
-        x: 0.5,
-        y: 1,
+        x: 0.52,
+        y: 0.9,
         fwhm: 0.08,
+        shape: { kind: 'lorentzian' } as Shape1D,
       },
     ];
-    let result = optimize(data, peakList, { shape: { kind: 'lorentzian' } });
+    let result = optimize(data, peakList);
```

Comment:
The former syntax should still be allowed and both test should be there, with the old syntax (all the shapes are the same) and with the new syntax in which you can define the shape at the level of each peak.